### ISSUE TITLE
Request for Experimental Seeds and Model Weights to Facilitate Replication of Bert-Base Results from Your Paper

### ISSUE THREAD

**Issue by JOHNNY-fans** [sh0416/clrcmd#7]
Hello, I am currently trying to replicate the experimental results presented in your paper. To maintain consistency, I set the seed to 42 in my experiments. However, I have noticed a significant discrepancy in the results when using the bert-base model, compared to what was reported in your paper, despite not altering any other parameters. To better understand this discrepancy, could you please share the four random seeds used in your experiments for the paper? Additionally, if it's convenient, could you also provide the model weights used in your experiments? Thank you very much for your assistance and willingness to share.

**Comment by sh0416** [comment]
Thanks for reaching our project.

Some implementation details are changed in this codebase.

Fortunately, I have a checkpoint that is used to report the experimental result.

I will share the checkpoint after the inspection for your stable usage. So, please wait a minute.

**Comment by sh0416** [comment]
I've just followed the process written in README.md.

I think the link provided in the README.md share the checkpoint.

Also, the checkpoint shows the same performance when we follow the README.md.

If you could not download the checkpoint, use this [link](https://drive.google.com/drive/folders/1P_COPucV-japQcrYrkWuMk_r24Cf9ca-?usp=sharing).

<img width="551" alt="image" src="https://github.com/sh0416/clrcmd/assets/12251974/85ea12fc-d70f-4af8-b301-fac701cb8a42">

**Comment by JOHNNY-fans** [comment]
Thank you very much for your assistance. I have successfully replicated the experimental results from your paper. However, I have some questions regarding the specific operation of the four random seeds mentioned in the article, as it currently seems like there is only one checkpoint. Additionally, for the more precise continuation of my experiments, I would like to know if it's possible to obtain the random seed corresponding to the current optimal checkpoint. This would enable me to better follow your experimental setup for my research. Thank you very much for your support and sharing. (My current experimental results suggest that the impact of the random seed seems to exceed the acceptable range.)"

**Comment by JOHNNY-fans** [comment]
I also noticed in the code repository that the seeds seem to be set to 0, 2, 3, and 4. Could you please confirm if these are indeed the random seeds used for the experiments?

**Comment by sh0416** [comment]
From my old memory, I think I did not consider sophisticated adjustment about the random seed.
The more important factor I remember is the development environment such as nvidia driver version or cuda version.
I don't know why, but expect that there exists an instability during my research.
I was done in 2021 fall, so maybe the engineering stack might be prematured in that time.

Hope it helps.

**Comment by JOHNNY-fans** [comment]
Thank you for your reply, if you still have the code from your experiments, would you be so kind as to help me determine the seed, it's fine if it matches your experiments, and it would be best if you have the trainer_state.json, I'm currently implementing it on 3090, I'll give it another go.

**Comment by sh0416** [comment]
As far as I know, the perfect replication of training experiments is extremely hard due to the hardware constraints.

FYI, I attach the pytorch document about randomness, and, to the best of my knowledge, the results of some ops become different even if we set the same seed when we use different machines and cuda versions and so forth.

https://pytorch.org/docs/stable/notes/randomness.html

And I remember the variance would exist in the multiple trials, but the correlation between valid dataset and test datasets exists, so selecting the best model would be possible in that time.

**Comment by sh0416** [comment]
I found the checkpoints that I've trained. I will sort out these checkpoints and release them within near future. There exist training states that you want, and the best seed in my setting is 3, but I couldn't guarantee that it would be optimal in your environment.

**Comment by JOHNNY-fans** [comment]
thanks for your patient reply.

**Comment by sh0416** [comment]
FYI, I released the checkpoints in the huggingface repository.

https://huggingface.co/sh0416/clrcmd

Unfortunately, I removed the checkpoint that is not best at the past, so I couldn't recover it.

You can check the training state in the repository and take brief hyperparameter settings in the following figure.

Also, I've lost the roberta checkpoint of rcmd version (20210930), which is trained earlier, but there exists a recent version (20211030_191325) that shows competitive performance, so check it out.

<img width="1289" alt="image" src="https://github.com/sh0416/clrcmd/assets/12251974/71cc68a9-0622-4d16-863f-cfb60ae2cb3d">
